core/admin.py

```diff
@@ -1,11 +1,44 @@
 from django.contrib import admin
 from django_admin_display import admin_display
 from django.utils.safestring import mark_safe
+from django.db.models import FileField
+import os
 
 from . import models
 from . import tasks
 
 
+def _text_preview(log_file: FileField):
+    """
+    Return the text of a file if it is short or the last portion of it if it is long.
+
+    :params: log_file A FileField to read text from.
+    """
+    # max file size for display, currently 10kb
+    maxlen = 10000
+    if log_file:
+        with log_file.open('rb') as datafile:
+            if len(datafile) > 0:
+                try:
+                    datafile.seek(-maxlen, os.SEEK_END)
+                except OSError as exc:
+                    if exc.errno != 22:
+                        # reraise exceptions except for trying to seek before the beginning
+                        raise
+                message = datafile.read().decode(errors='replace')
+                if len(log_file) < maxlen:
+                    return message
+                else:
+                    prefix_message = f"""The output is too large to display in the browser.
+                        Only the last {maxlen} characters are displayed.
+                        """
```

Comment:
In triple-quote string formatting, white-space is preserved, so we want don't want spaces on the left for subsequent lines (since it will indent that line in the display).

To match the formatting you had earlier, we also want a blank line before the closing triple quotes.

You can see how it looks in the application by temporarily setting the `maxlen` threshold to a small number.